src/attestation_types/sigdata.rs

```diff
@@ -0,0 +1,234 @@
+        r.copy_from_slice(&bytes[0..32]);
+
+        let mut s = [0u8; 32];
+        s.copy_from_slice(&bytes[32..64]);
+
+        Self { r, s }
+    }
+}
+
+/// EC KT-I Public Key, the x-coordinate followed by
+/// the y-coordinate (on the RFC 6090P-256 curve),
+/// 2 x 32 bytes.
+/// A.4, Table 7
+#[derive(Default, Clone, Copy)]
+#[repr(C)]
+pub struct ECDSAPubKey {
+    /// x coordinate
+    pub x: [u8; 32],
+
+    /// y coordinate
+    pub y: [u8; 32],
+}
+
+impl From<&[u8; 64]> for ECDSAPubKey {
+    fn from(bytes: &[u8; 64]) -> Self {
+        let mut x = [0u8; 32];
+        x.copy_from_slice(&bytes[0..32]);
+
+        let mut y = [0u8; 32];
+        y.copy_from_slice(&bytes[32..64]);
+
+        Self { x, y }
+    }
+}
+
+/// Section A.4, Table 9
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[repr(u16)]
+pub enum CertDataType {
+    /// Byte array that contains concatenation of PPID, CPUSVN,
+    /// PCESVN (LE), PCEID (LE)
+    PpidPlaintext = 1,
+
+    /// Byte array that contains concatenation of PPID encrypted
+    /// using RSA-2048-OAEP, CPUSVN,  PCESVN (LE), PCEID (LE)
+    PpidRSA2048OAEP = 2,
+
+    /// Byte array that contains concatenation of PPID encrypted
+    /// using RSA-3072-OAEP, CPUSVN, PCESVN (LE), PCEID (LE)
+    PpidRSA3072OAEP = 3,
+
+    /// PCK Leaf Certificate
+    PCKLeafCert = 4,
+
+    /// Concatenated PCK Cert Chain  (PEM formatted).
+    /// PCK Leaf Cert||Intermediate CA Cert||Root CA Cert
+    PCKCertChain = 5,
+
+    /// Intel SGX Quote (not supported).
+    Quote = 6,
+
+    /// Platform Manifest (not supported).
+    Manifest = 7,
+}
+
+impl Default for CertDataType {
+    fn default() -> Self {
+        Self::PCKLeafCert
+    }
+}
+
+impl TryFrom<u16> for CertDataType {
+    type Error = QuoteError;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        match value {
+            1 => Ok(CertDataType::PpidPlaintext),
+            2 => Ok(CertDataType::PpidRSA2048OAEP),
+            3 => Ok(CertDataType::PpidRSA3072OAEP),
+            4 => Ok(CertDataType::PCKLeafCert),
+            5 => Ok(CertDataType::PCKCertChain),
+            6 => Ok(CertDataType::Quote),
+            7 => Ok(CertDataType::Manifest),
+            _ => Err(QuoteError(format!("Unknown Cert Data type: {}", value))),
+        }
+    }
+}
+
+impl fmt::Display for CertDataType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            CertDataType::PpidPlaintext => write!(f, "PpidPlaintext"),
+            CertDataType::PpidRSA2048OAEP => write!(f, "PpidRSA2048OAEP"),
+            CertDataType::PpidRSA3072OAEP => write!(f, "PpidRSA3072)AEP"),
+            CertDataType::PCKLeafCert => write!(f, "PCKLeafCert"),
+            CertDataType::PCKCertChain => write!(f, "PCKCertChain"),
+            CertDataType::Quote => write!(f, "Quote"),
+            CertDataType::Manifest => write!(f, "Manifest"),
+        }
+    }
+}
+
+/// A.4, Table 4
+#[derive(Default)]
+#[repr(C)]
```

Comment:
If it can't be `#[repr(c)]` then I may need to have a different version of the struct that's created from this original once the size is known. Maybe a debt issue...

Reply:
Why do you need `#[repr(C)]` ? for FFI?

Reply:
Yeah, this may potentially get passed around between Rust and C.

Reply:
It may also be possible in the near future to remove the `vec` fields, which would be ideal. Although the `Quote` can change in size, I think the version we're using will always be the same size. But I have to test that theory.

Reply:
@haraldh I made a debt issue here: https://github.com/enarx/sgx/issues/16